src/hypothesis/internal/charmap.py

```diff
@@ -141,6 +141,40 @@ def _union_intervals(x, y):
     return tuple(result)
 
 
+def _subtract_intervals(x, y):
+    """Return a list of intervals that bounds everything bounded by x that is
+    not also bounded by y."""
+    if not y:
```

Comment:
This function is quite cryptic. Could there be a short notice what happens here?

Reply:
Yeah, I’m not entirely sure what this function is meant to do. Maybe an example would be helpful?

Reply:
Better?

Reply:
Awesome! Thank you!